File: src/test/java/net/slipp/web/user/ApiUsersControllerTest.java
package net.slipp.web.user;

import static org.hamcrest.CoreMatchers.*;
import static org.junit.Assert.*;
import net.slipp.domain.user.SocialUser;

import org.junit.Before;
import org.junit.Test;

public class ApiUsersControllerTest {
	private ApiUsersController dut;
	
	@Before 
	public void setup() {
		dut = new ApiUsersController();
	}
	
	@Test
	public void checkDuplicate_doesnot_existed() {
		String actual = dut.checkDuplicate(SocialUser.GUEST_USER, null);
		assertThat(actual, is("false"));
	}

	@Test
	public void checkDuplicate_login_isSameUser() {
		SocialUser loginUser = new SocialUser(1L);
		SocialUser existedUser = loginUser;
		String actual = dut.checkDuplicate(loginUser, existedUser);
		assertThat(actual, is("false"));
	}
	
	@Test
	public void checkDuplicate_login_isNotSameUser() {
		SocialUser loginUser = new SocialUser(1L);
		SocialUser existedUser = new SocialUser(2L);
		String actual = dut.checkDuplicate(loginUser, existedUser);
		assertThat(actual, is("true"));
	}
}

File: src/test/scala/net/slipp/web/user/ApiUsersControllerTest.scala
package net.slipp.web.user

import net.slipp.domain.user.SocialUser

import org.junit.Before
import org.junit.Test
import org.junit.Assert._
import org.hamcrest.CoreMatchers._

class ApiUsersControllerTest {
  private var dut: ApiUsersController = null
  
  @Before 
  def setup() {
    dut = new ApiUsersController()
  }
  
  @Test
  def checkDuplicate_doesnot_existed() {
    val actual = dut.checkDuplicate(SocialUser.GUEST_USER, null)
    assertThat(actual, is(false))
  }
  
  @Test
  def checkDuplicate_login_isSameUser() {
    val loginUser = new SocialUser(1L)
    val existedUser = loginUser
    val actual = dut.checkDuplicate(loginUser, existedUser)
    assertThat(actual, is(false))
  }
  
  @Test
  def checkDuplicate_login_isNotSameUser() {
    val loginUser = new SocialUser(1L)
    val existedUser = new SocialUser(2L)
    val actual = dut.checkDuplicate(loginUser, existedUser)
    assertThat(actual, is(true))
  }
}